keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -104,7 +112,12 @@ static INDArray unsafeGetNd4J(DoubleTensor that) {
         if (that.isLengthOne()) {
             return TypedINDArrayFactory.scalar(that.scalar(), BUFFER_TYPE).reshape(that.getShape());
         }
-        return ((Nd4jDoubleTensor) that).tensor;
+
+        if (that instanceof Nd4jDoubleTensor) {
+            return ((Nd4jDoubleTensor) that).tensor;
```

Comment:
What's the point of this?

Reply:
It's extracting the NDArray bit out of an ND4J tensor - but given we can now have JVMDoubleTensors - there's some weirdness there we've got to work around.